go/logic/applier.go

```diff
@@ -934,6 +937,42 @@ func (this *Applier) CreateAtomicCutOverSentryTable() error {
 	return nil
 }
 
+// WaitForAtomicCutOverRename waits for the cut-over RENAME operation while periodically
+// pinging the applier connection to avoid the connection becoming idle.
+func (this *Applier) WaitForAtomicCutOverRename(tx *gosql.Tx, okToUnlockTable <-chan bool) {
+	ticker := time.NewTicker(time.Millisecond * 500)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-okToUnlockTable:
+			this.migrationContext.Log.Infof("Will now proceed to drop magic table and unlock tables")
+			return
+		case <-ticker.C:
+			// keep connection alive as we wait for the RENAME.
+			if _, err := tx.Exec(`select /* gh-ost */ 1`); err != nil {
+				this.migrationContext.Log.Errorf("Failed to ping applier connection: %v", err)
+			}
+		}
+	}
+}
+
+// InitAtomicCutOverIdleTimeout sets the cut-over session wait_timeout, returning a func() to rollback
+// to the original wait_timeout and an error, if any.
+func (this *Applier) InitAtomicCutOverIdleTimeout(tx *gosql.Tx) (restoreTimeoutFunc func(), err error) {
+	this.migrationContext.Log.Infof("Setting cut-over idle timeout as %d seconds", this.migrationContext.CutOverIdleTimeoutSeconds)
+	query := fmt.Sprintf(`set /* gh-ost */ session wait_timeout:=%d`, this.migrationContext.CutOverIdleTimeoutSeconds)
+	_, err = tx.Exec(query)
+	return func() {
```

Comment:
The generated function doesn't use anything special coming from `InitAtomicCutOverIdleTimeout`, ie no local variables etc. As such, it will be simpler for `InitAtomicCutOverIdleTimeout` to only return `error`, and for the calling code (`AtomicCutOverMagicLock()`, below) to just `defer sqlutils.ExecNoPrepare(this.db, <restore wait time>)`)

Reply:
Makes sense. Updated 👍